### PR TITLE
Fixes for Swift 5.3, Alamofire 5.2.2 - SPM + Carthage

### PR DESCRIPTION
**Pull request by rursache**
After a lot of playing around with SPM's Package.swift, i managed to get it working, ready to be merged.

Carthage support was added by @ozgur

### CODE REVIEW

**Comment by mxcl** [comment]
Sorry, but as mentioned in #44, #45, #43 and some closed tickets, we cannot update Alamofire without breaking API compatibility.

Although at this point I'm starting to think: fuck it.

v7 will drop bundling the extensions with the main “pod” thus allowing them to be versioned separately.

**Comment by mxcl** [comment]
I think what we’ll do is release this for Carthage/SwiftPM and CocoaPods will have to suffer for a bit.

**Comment by mxcl** [comment]
We have to merge one of these but I don’t really know how to go about it due to the concerns listed in #44

**Comment by mxcl** [comment]
I think we can just add a new subspec and submodule to PMK master.

File: Sources/Alamofire+Promise.swift [hunk 1]
```diff
@@ -1,189 +1,176 @@
...
-                case .success(let value):
-                    seal.fulfill((value, PMKAlamofireDataResponse(response)))
-                case .failure(let error):
-                    seal.reject(error)
-                }
-            }
-        }
-    }
+	/// Adds a handler to be called once the request has finished.
+	public func response(_: PMKNamespacer, queue: DispatchQueue = .main) -> Promise<(URLRequest, HTTPURLResponse, Data)> {
```
Comment (mxcl) [inline Sources/Alamofire+Promise.swift:19]: These whitespace changes are naughty.

**Comment by mxcl** [comment]
We don’t have CI (Travis no longer provides OSS CI) so I will merge if you give your word that this works…?

**Comment by rursache** [comment]
> We don’t have CI (Travis no longer provides OSS CI) so I will merge if you give your word that this works…?

i'm using it for months already using my fork, everything works if integrated via SPM. haven't tested Cocoapods

**Pull request merged**